tests/unit/test_application.py

```diff
@@ -450,6 +450,35 @@ def test_show_app_name_and_version(monkeypatch, capsys, app):
     assert f"Starting testcraft, version {app.app.version}" in err
 
 
+@pytest.mark.parametrize("verbosity", list(craft_cli.EmitterMode))
+def test_set_verbosity_from_env(monkeypatch, capsys, app, verbosity):
+    """Test that the emitter verbosity is set from the environment."""
+    monkeypatch.setattr(sys, "argv", ["testcraft"])
+    monkeypatch.setenv("CRAFT_VERBOSITY_LEVEL", verbosity.name)
+
+    with pytest.raises(SystemExit):
+        app.run()
+
+    _, err = capsys.readouterr()
+    assert "testcraft [help]" in err
+    assert craft_cli.emit._mode == verbosity
```

Comment:
can you use `get_mode()` here instead?

Reply:
Oh, I just remembered why I used `_mode`, the emitter is stopped at that point.
```
RuntimeError: Emitter is stopped already
```

Reply:
ah I see, that's fine then